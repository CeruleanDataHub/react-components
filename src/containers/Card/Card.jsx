import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import { Icon, iconNameProps } from "../Icon/Icon.jsx";

const CardDash = styled.div`
  margin: "2em";
  border-radius: 0.5em;
  border: 2px dashed #ffffff;
`;
const CardContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: ${(props) => props["justify-content"] || "center"};
  margin: 1em;
  border-radius: 0.25em;
  border: 1px solid #000000;
  background-color: #ffffff;
  height: ${(props) => props.height || "20em"};
  width: ${(props) => props.width || "auto"};
  max-width: ${(props) => props["max-width"] || "auto"};
  font-family: "Open Sans", sans-serif;
  text-align: ${(props) => props["text-align"] || "center"};
  padding: 1em;
  box-sizing: border-box;
`;
const CardHeader = styled.h3`
  font-family: "Exo 2", sans-serif;
  font-weight: 400;
  text-align: center;
  text-transform: uppercase;
`;
const CardBody = styled.div`
  text-align: center;
`;

export const Card = ({ title, icon, children }) => (
  <CardDash>
    <CardContainer>
      {icon && <Icon name={icon} />}
      {title && <CardHeader>{title}</CardHeader>}
      <CardBody>{children}</CardBody>
    </CardContainer>
  </CardDash>
);

Card.propTypes = {
  title: PropTypes.string,
  icon: iconNameProps,
  children: PropTypes.node,
};

Card.defaultProps = {
  title: null,
  icon: null,
  children: "",
};
