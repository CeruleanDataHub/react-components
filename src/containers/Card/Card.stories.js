import React from 'react';

import { Card } from './Card';

export default {
  title: 'Card',
  component: Card,
};

export const CardStory = () => (
  <div style={{ width: '500px' }}>
    <Card item={{ iconClass: 'users', title: 'Users' }}>You have 30 users</Card>
  </div>
);
