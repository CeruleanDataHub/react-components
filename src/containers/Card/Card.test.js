import React from "react";
import renderer from "react-test-renderer";

import { Card } from "./Card";

describe("Card", () => {
  it("should render card", () => {
    const component = renderer.create(<Card />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render card with item", () => {
    const component = renderer.create(
      <Card item={{ iconClass: "lnr-users", title: "Users" }} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render card with children", () => {
    const component = renderer.create(<Card>Children textual content</Card>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
