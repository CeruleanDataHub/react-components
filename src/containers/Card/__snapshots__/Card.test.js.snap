// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders 1`] = `
<Card
  icon={null}
  title={null}
>
  <styled.div>
    <div
      className="sc-AxiKw dQZYkP"
    >
      <styled.div
        data-card-body-test={true}
      >
        <div
          className="sc-AxhUy fNfuCs"
          data-card-body-test={true}
        >
          Some children
        </div>
      </styled.div>
    </div>
  </styled.div>
</Card>
`;
